### PR TITLE
test: add WorkPlanByTestController integration tests with dataset

### PR DESCRIPTION
**Pull request by Rakesh-honawad**
## Description

This PR adds integration tests for the WorkPlanByTestController.

### Changes:
- Added `WorkPlanByTestControllerTest` to cover controller endpoints
- Created dataset `workplan-by-test-controller.xml` for test setup
- Updated `AppTestConfig` to include the new dataset

### Test Coverage:
- Valid request scenarios
- Edge cases (empty results)
- Data setup validation

### Notes:
- All relevant workplan tests pass locally
- Branch is rebased on latest `develop`
- Spotless formatting applied

## Impact
Improves test coverage for workplan controller functionality and ensures stability of related endpoints.

### CODE REVIEW

File: src/test/java/org/openelisglobal/workplan/controller/WorkPlanByTestControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
+package org.openelisglobal.workplan.controller;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (Agaba-derrick) [inline src/test/java/org/openelisglobal/workplan/controller/WorkPlanByTestControllerTest.java]: Can we avoid  Wild cards ?

**Review by Agaba-derrick — CHANGES_REQUESTED** [review]

File: src/test/java/org/openelisglobal/workplan/controller/WorkPlanByTestControllerTest.java (hunk 1 above)
Comment (Rakesh-honawad) [reply src/test/java/org/openelisglobal/workplan/controller/WorkPlanByTestControllerTest.java]: Hi @Agaba-derrick,

Thank you for the feedback! You're right  wildcard 
imports make it harder to see exactly what's being used.
I'll replace all wildcard static imports with explicit 
ones. Will push the fix shortly!

Comment (Rakesh-honawad) [reply src/test/java/org/openelisglobal/workplan/controller/WorkPlanByTestControllerTest.java]: Done - replaced all wildcard static imports with explicit imports. Fix pushed in the latest commit."

**Comment by mozzy11** [comment]
Thank you for this contribution, and apologies that it sat unreviewed for so long.

We are doing a cleanup of the PR queue: this PR is now more than 3 months old and has unresolved merge conflicts with develop. The codebase has changed substantially in that time (the frontend moved to Vite and is being migrated to TypeScript, the test infrastructure was rewritten, and several modules — test catalog, analyzers, EQA, storage — were reworked), so a mechanical rebase is unlikely to succeed.

Closing as stale per our queue policy. **If the problem you fixed still exists on current develop, please open a fresh PR against the latest code — we are actively reviewing now and it will get prompt attention.** Thanks again for contributing to OpenELIS!
